packages/ra-core/src/controller/useListParams.ts

```diff
@@ -135,7 +139,9 @@ const useListParams = ({
         perPage,
     ];
 
-    const queryFromLocation = parseQueryFromLocation(location);
+    const queryFromLocation = syncWithLocation
+        ? parseQueryFromLocation(location)
```

Comment:
the location is grabbed in `useListController` and passed to `useListParams`. I think we'd better call `useLocation` directly in `useListParams`.

Reply:
Done but not visible on this comment